tests/Unit/Application/Commands/DeleteShoppingItem/Validation/Rules/ShoppingItemAlreadyDeletedTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Tests\Unit\Application\Commands\DeleteShoppingItem\Validation\Rules;
+
+use Lindyhopchris\ShoppingList\Application\Commands\DeleteShoppingItem\DeleteShoppingItemModel;
+use Lindyhopchris\ShoppingList\Application\Commands\DeleteShoppingItem\Validation\Rules\ShoppingItemAlreadyDeleted;
+use Lindyhopchris\ShoppingList\Domain\ShoppingItemSelector;
+use Lindyhopchris\ShoppingList\Domain\ShoppingItemStack;
+use Lindyhopchris\ShoppingList\Domain\ShoppingList;
+use Lindyhopchris\ShoppingList\Persistance\ShoppingListRepositoryInterface;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class ShoppingItemAlreadyDeletedTest extends TestCase
+{
+    /**
+     * @var ShoppingListRepositoryInterface|MockObject
+     */
+    private ShoppingListRepositoryInterface|MockObject $repository;
+
+    /**
+     * @var ShoppingItemAlreadyDeleted
+     */
+    private ShoppingItemAlreadyDeleted $rule;
+
+    /**
+     * @return void
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->rule = new ShoppingItemAlreadyDeleted (
+            $this->repository = $this->createMock(ShoppingListRepositoryInterface::class),
+        );
+    }
+
+    public function testItemDoesNotExist(): void
```

Comment:
Good job on bringing this test in too.

Required: at the moment we're only testing one scenario here, which is why we haven't picked up on a bug in our rule. The scenarios that should be tested are all the combinations of what could happen:

1. Shopping list and item do exist (the rule returns no errors)
2. Shopping list does not exist (depending on how you fix this based on my previous comment, the rule should either return no errors assuming the other rule will run, or can add an error itself).
3. Shopping item does not exist (this is the test you've already written).

If you take a look at `ShoppingItemExistsAndIsNotTickedOffTest` (in the `TickOffShoppingItem` tests) you can see those three combinations being tested.